From 6fd7d5b0ad95f576a1359b47102d269b6d634a81 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 20 Nov 2019 17:06:25 +0200
Subject: [PATCH] [SW-0] gaudi: vuart: add support for preboot_putc_virt_uart()

It was just an empty stub till now
Turn it to be like Goya

Change-Id: I86b518ac2d11575f9dec8fd7009ba1ce8d31ffb2
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 48b82f8..abcbcb4 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -161,7 +161,19 @@
 }
 
 void preboot_delay(void) {}
-void preboot_putc_virt_uart(const char c) {}
+
+#define mmPCIE_WRAP_VUART_TX_DATA	0xC01F60
+#define mmPCIE_WRAP_VUART_TX_STATUS	0xC01F64
+void preboot_putc_virt_uart(const char c)
+{
+	/*
+	 * update vUART to HOST
+	 * DATA: mmPCIE_WRAP_VUART_TX_DATA
+	 * STATUS: mmPCIE_WRAP_VUART_TX_STATUS
+	 */
+	regmap_writel(c, mmPCIE_WRAP_VUART_TX_DATA);
+	regmap_writel(1, mmPCIE_WRAP_VUART_TX_STATUS);
+}
 
 u32 get_cs_pad_sel(void)
 {
